exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/DrillFunctionRegistry.java

```diff
@@ -218,4 +302,141 @@ private void registerOperatorsWithoutInference(DrillOperatorTable operatorTable)
       }
     }
   }
+
+  /**
+   * Function registry holder. Stores function implementations by jar name, function name.
+   */
+  private class GenericRegistryHolder<T, U> {
```

Comment:
How is this class tested? Being a private, inner class, it is impossible to unit test. Not convinced adequate testing can be done by testing Drill as a whole via SQL statements. This may be a project philosophy issue if we don't require "unit" tests in the traditional meaning: tests of specific modules outside the context of the entire system.